contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -201,10 +200,12 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
 
         let unregistered_token_id = token_id::ensure_token_not_registered(env, token_id.clone())?;
 
-        let token_address = Self::deploy_token(env, unregistered_token_id, token_metadata, minter)?;
+        let _ = Self::deploy_token(env, unregistered_token_id.clone(), token_metadata, minter)?;
 
         if initial_supply > 0 {
-            StellarAssetClient::new(env, &token_address).mint(&caller, &initial_supply);
+            let token_config_value = Self::token_id_config(env, unregistered_token_id.into())?;
```

Comment:
```suggestion
            let token_config_value = Self::token_id_config(env, token_id)?;
```
don't we already have a `token_id` variable?